tests/crypto_crls_cert_chains/host/CMakeLists.txt

```diff
@@ -3,12 +3,86 @@
+            # Creating Leaf certificates signed by the Intermediate CA
+            COMMAND sleep 1
+            COMMAND echo 'Creating Leaf certificates signed by the Intermediate CA....'
+            COMMAND openssl genrsa -out ${DATA_DIR}/Leaf1.key.pem
+            COMMAND openssl req -new -key ${DATA_DIR}/Leaf1.key.pem -out ${DATA_DIR}/Leaf1.csr -subj "/C=US/ST=Ohio/L=Columbus/O=Acme Company/OU=Acme/CN=Leaf1"
+            COMMAND openssl x509 -req -in ${DATA_DIR}/Leaf1.csr -CA ${DATA_DIR}/IntermediateCA1.crt.pem -CAkey ${DATA_DIR}/IntermediateCA1.key.pem -CAcreateserial -out ${DATA_DIR}/Leaf1.crt.pem -days 3650
+
+            COMMAND sleep 1
+            COMMAND openssl genrsa -out ${DATA_DIR}/Leaf2.key.pem
+            COMMAND openssl req -new -key ${DATA_DIR}/Leaf2.key.pem -out ${DATA_DIR}/Leaf2.csr -subj "/C=US/ST=Ohio/L=Columbus/O=Acme Company/OU=Acme/CN=Leaf2"
+            COMMAND openssl x509 -req -in ${DATA_DIR}/Leaf2.csr -CA ${DATA_DIR}/IntermediateCA1.crt.pem -CAkey ${DATA_DIR}/IntermediateCA1.key.pem -CAcreateserial -out ${DATA_DIR}/Leaf2.crt.pem -days 3650
+
+            # Setup Certificate Revocation lists for the following test cases
+            #  root_crl1 and root_crl2 are issued by root.
+            #  root_crl1 revokes no certificates.
+            #  root_crl2 revokes intermediate cert.
+            #  intermediate_crl1 and intermediate_crl2 are issued by intermediate.
+            #  intermediate_crl1 revokes no certificates.
+            #  intermediate_crl2 revokes leaf2.
+
+            COMMAND touch root_index.txt
+            COMMAND touch root_index2.txt
+            COMMAND touch intermediate_index.txt
+            COMMAND touch intermediate_index2.txt
+            COMMAND echo "00" > root_crl_number
+            COMMAND echo "00" > root_crl_number2
+            COMMAND echo "00" > intermediate_crl_number
+            COMMAND echo "00" > intermediate_crl_number2
+
+            # Generating a root and intermediate CRLs
+            COMMAND openssl ca -gencrl -config ${DATA_DIR}/root.cnf -out ${DATA_DIR}/root_crl1.pem 
+            COMMAND openssl ca -gencrl -config ${DATA_DIR}/root2.cnf -out ${DATA_DIR}/root_crl2.pem> /dev/null
+            COMMAND openssl ca -gencrl -config ${DATA_DIR}/intermediate.cnf -out ${DATA_DIR}/intermediate_crl1.pem
+            COMMAND openssl ca -gencrl -config ${DATA_DIR}/intermediate2.cnf -out ${DATA_DIR}/intermediate_crl2.pem
+
+            # Revoking intermediate and leaf certs
+            COMMAND openssl ca -revoke ${DATA_DIR}/IntermediateCA1.crt.pem -keyfile ${DATA_DIR}/RootCA1.key.pem -cert ${DATA_DIR}/RootCA1.crt.pem -config ${DATA_DIR}/root2.cnf
+            COMMAND openssl ca -revoke ${DATA_DIR}/Leaf2.crt.pem -keyfile ${DATA_DIR}/IntermediateCA1.key.pem -cert ${DATA_DIR}/IntermediateCA1.crt.pem -config ${DATA_DIR}/intermediate2.cnf
+
+            # recreate updated CRL lists
+            COMMAND openssl ca -gencrl -config ${DATA_DIR}/root2.cnf -out ${DATA_DIR}/root_crl2.pem
```

Comment:
The root2.cnf and intermediate2.cnf seem unnecessary.
It seems like you can pass root.cnf and intermediate.cnf here and it should still work.

The crl_number and the index.txt files specified in the cnf act as the database.

Reply:
I tried that and it involved deleting and recreating  index.txt for each crl creation, which makes it hard to understand CRL creation process. That's why I decided to totally separate the generation of those two CRLs.